src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -67,6 +80,50 @@ public FilterPanel(UI ui, IModel model, PanelControl parentPanelControl, int pan
 
         filterTextField.setKeywords(all);
     };
+    
+    private Node createNameBar() {
+        nameText = new Text(panelName);
+    
+        renameButton = new Label(RENAME_PANEL);
+        renameButton.getStyleClass().add("label-button");
+        renameButton.setId(model.getDefaultRepo() + "_col" + panelIndex + "_renameButton");
+        renameButton.setOnMouseClicked(e -> {
+            TextInputDialog renameDialog = new TextInputDialog(panelName);
+            renameDialog.getEditor().setId("panelrenameinput");
+            renameDialog.setTitle("Rename " + panelName);
+            renameDialog.setHeaderText("Enter a new name for this panel.");
+            Optional<String> result = renameDialog.showAndWait();
+            mainStage.show();
+            String newName = "";
+            if (result.isPresent()) {
+                newName = result.get();
+            }
+            if (!newName.equals("")) {
+                panelName = newName;
+            }
+            if (panelName.length() > MAX_NAME_LENGTH) {
+                panelName = panelName.substring(0, MAX_NAME_LENGTH);
+            }
+            
```

Comment:
This is unnecessarily complicated. Can probably do something along the lines of `result.orElse("").substring(0, MAX_NAME_LENGTH)`.

The substring thing can also be factored out into a utility function (`truncate`?). It could add ellipses if the input is too long.